src/main/kotlin/org/neo4j/graphql/handler/QueryHandler.kt

```diff
@@ -77,6 +80,18 @@ class QueryHandler private constructor(
         val ordering = orderBy(variable, field.arguments)
         val skipLimit = SkipLimit(variable, field.arguments).format()
 
+        if ((env.getContext() as? QueryContext)?.optimizedQuery == true) {
+
+            val (partialQuery, filterParams) = OptimizedFilterHandler(type).generateFilterQuery(variable, field)
+            val statement = partialQuery
+                ?.returningDistinct(PassThrough("${mapProjection.query} AS $variable$ordering${skipLimit.query}"))
+                ?.build()
+                    ?: throw OptimizedQueryException()
+            val query = Renderer.getDefaultRenderer().render(statement)
```

Comment:
should that be something that's shared? perhaps as a config/injection

Reply:
It's the only place where it is used, no need to share it by now